server/lib/notifications/agents/webpush.ts

```diff
@@ -0,0 +1,205 @@
+import { getRepository } from 'typeorm';
+import webpush from 'web-push';
+import { hasNotificationType, Notification } from '..';
+import { MediaType } from '../../../constants/media';
+import { User } from '../../../entity/User';
+import { UserPushSubscription } from '../../../entity/UserPushSubscription';
+import logger from '../../../logger';
+import { Permission } from '../../permissions';
+import { getSettings, NotificationAgentConfig } from '../../settings';
+import { BaseAgent, NotificationAgent, NotificationPayload } from './agent';
+
+interface PushNotificationPayload {
+  notificationType: string;
+  mediaType?: 'movie' | 'tv';
+  tmdbId?: number;
+  subject: string;
+  message?: string;
+  image?: string;
+  actionUrl?: string;
+  requestId?: number;
+}
+
+class WebPushAgent
+  extends BaseAgent<NotificationAgentConfig>
+  implements NotificationAgent {
+  protected getSettings(): NotificationAgentConfig {
+    if (this.settings) {
+      return this.settings;
+    }
+
+    const settings = getSettings();
+
+    return settings.notifications.agents.webpush;
+  }
+
+  private getNotificationPayload(
+    type: Notification,
+    payload: NotificationPayload
+  ): PushNotificationPayload {
+    switch (type) {
+      case Notification.TEST_NOTIFICATION:
+        return {
+          notificationType: Notification[type],
+          subject: payload.subject,
+          message: payload.message,
+        };
+      case Notification.MEDIA_APPROVED:
+        return {
+          notificationType: Notification[type],
+          subject: payload.subject,
+          message: 'Your request has been approved.',
+          image: payload.image,
+          mediaType: payload.media?.mediaType,
+          tmdbId: payload.media?.tmdbId,
+          requestId: payload.request?.id,
+          actionUrl: `/${payload.media?.mediaType}/${payload.media?.tmdbId}`,
+        };
+      case Notification.MEDIA_AUTO_APPROVED:
+        return {
+          notificationType: Notification[type],
+          subject: payload.subject,
+          message: `${
+            payload.media?.mediaType === MediaType.MOVIE ? 'Movie' : 'Series'
+          } request was auto-approved.\nRequested by: ${
+            payload.request?.requestedBy.displayName
+          }`,
```

Comment:
```suggestion
          message: `Automatically approved a new ${
            payload.media?.mediaType === MediaType.MOVIE ? 'movie' : 'series'
          } request from ${payload.request?.requestedBy.displayName}.`,
```
I still strongly feel that we should not have a newline in this notification message.  We cannot guarantee that the first line will not wrap, and when it does, it will just result in the "requested by" information not being visible at all if only two lines of text are allowed.